File: Programmers/Swift/행렬의 곱셈.swift
import Foundation

func solution(_ arr1:[[Int]], _ arr2:[[Int]]) -> [[Int]] {
    
    var res = [[Int]](repeating: [Int](repeating:0, count: arr2[0].count), count: arr1.count)
    
    for y in 0..<res.count {
        for x in 0..<res[0].count {
            for i in 0..<arr1[0].count {
                res[y][x] += arr1[y][i]*arr2[i][x]
            }
        }
    }
    
    return res
}

File: Programmers/C++/행렬의 곱셈.cpp
#include <string>
#include <vector>
#define FOR(i,n) for(int i=0;i<(n);++i)
#define FORS(i,s,n) for(int i=(s);i<(n);++i)
using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    
    vector<vector<int>> res;
    
    res.resize(arr1.size());
    FOR(i,res.size()) res[i].resize(arr2[0].size());
    
    FOR(y,res.size()) FOR(x,res[0].size()) {
        FOR(i,arr1[0].size()) {
            res[y][x]+= arr1[y][i]*arr2[i][x];
        }
    }
    
    return res;
}
